components/constraints.js

```diff
@@ -149,13 +149,22 @@ class Constraints {
   // Handles type constraint. Will override datatype if top level
   // type constraint.
   typeConstraint(constraint, subpath) {
+    let constraintName, href;
+
     const element = this.elements[constraint.fqn];
-    var constraintName = (this.field.valueType === 'RefValue') ? `ref(${element.name})` : element.name;
-    if (this.field.valueType === 'RefValue') {
-      constraintName = `${constraintName}`;
-    }
+    if (element) {
+      constraintName = (this.field.valueType === 'RefValue') ? `ref(${element.name})` : element.name;
+      if (this.field.valueType === 'RefValue') {
+        constraintName = `${constraintName}`; //TODO: This doesn't do anything...
```

Comment:
This obviously (as confirmed by the TODO) doesn't do anything -- but what is the intent?  Why is this code here at all?

Reply:
Honestly, no idea. It looks like it might've been leftover special logic code from when this tool was first written. 

I planned on removing it, although I wasn't sure if I should touch it during a bug fix (in case there's some reason for its existence), or if I should save it for a later code clean-up/refactor.

Reply:
I say kill it.

Reply:
Done.